### PR TITLE
Decord Audio

### PR DESCRIPTION
**Pull request by yinweisu**

### CODE REVIEW

File: python/decord/audio_reader.py [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+        self._num_samples_per_channel = _CAPI_AudioReaderGetNumSamplesPerChannel(self._handle)
+        self._num_channels = _CAPI_AudioReaderGetNumChannels(self._handle)
+        self.sample_rate = sample_rate
+
+    def __del__(self):
+        if self._handle:
+            _CAPI_AudioReaderFree(self._handle)
+
+    def __getitem__(self, idx):
+        """Get sample at `idx`.
```
Comment (zhreshold) [inline python/decord/audio_reader.py:61]: `idx` is the index of resampled audio, unit is `sample`

File: python/decord/audio_reader.py [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+        -------
+        double
+            Duration of the audio in secs.
+
+        """
+        return self._duration
+
+    def __get_num_padding(self):
+        """Get number of samples needed to pad the audio to start at time 0."""
+        self._num_padding = _CAPI_AudioReaderGetNumPaddingSamples(self._handle)
```
Comment (zhreshold) [inline python/decord/audio_reader.py:127]: can be improved by lazy caching the padding, e.g., 
```
if self._num_padding is None:
  self._num_padding = _CAPI_AudioReaderGetNumPaddingSamples(self._handle)
return self._num_padding
```

File: src/audio/audio_reader.h [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+//
+
+#ifndef DECORD_AUDIO_READER_H_
+#define DECORD_AUDIO_READER_H_
+
+#include <vector>
+
+#include "../../include/decord/audio_interface.h"
+#ifdef __cplusplus
+extern "C" {
```
Comment (zhreshold) [inline src/audio/audio_reader.h:12]: do we need this in header file? better moved to .cc file if not required here

File: tests/cpp/audio/test_ffmpeg_audio_reader.cc [hunk 4]
```diff
@@ -0,0 +1,12 @@
+//
+// Created by Yin, Weisu on 1/15/21.
+//
+
+#include <decord/audio_interface.h>
+#include <decord/base.h>
+
+
+int main() {
+    auto audioReader = decord::GetAudioReader("/Users/weisy/Developer/yinweisu/decord/tests/cpp/audio/test.mp3", -1, decord::kCPU, 0, 0);
```
Comment (zhreshold) [inline tests/cpp/audio/test_ffmpeg_audio_reader.cc:10]: use audio file in examples?

**Review by zhreshold — CHANGES_REQUESTED** [review]

File: python/decord/audio_reader.py (hunk 1 above)
Comment (yinweisu) [reply python/decord/audio_reader.py:61]: Do you mean I should change idx to be sample?

File: src/audio/audio_reader.h (hunk 3 above)
Comment (yinweisu) [reply src/audio/audio_reader.h:12]: Audio reader header is using type defined in swersample, so moving to cc won't compile. I moved this include to ffmpeg_common to keep ffmpeg lib organized.

**Review by zhreshold — APPROVED** [review]

**Pull request merged**